apps/lib/apps.c

```diff
@@ -1100,8 +1106,6 @@ int load_key_certs_crls(const char *uri, int format, int maybe_stdin,
         failed = FAIL_NAME;
         if (failed != NULL && !quiet)
             BIO_printf(bio_err, "Could not find");
-    } else if (!quiet) {
-        BIO_printf(bio_err, "Could not read");
     }
```

Comment:
I don't understand this while if statement.  Like, can we ever *not* fail?

Reply:
Yeah, this code is quite convoluted and confusing - the FAIL_NAME macro above can actually evaluate to NULL.

Reply:
I agree that the code is (needlessly) confusing, at least on the variable `failed`.
In the first half of the function, it is governed by the `FAIL_NAME` macro,
but in the 2nd half it has a slightly different meaning - and therefore should better have a different name.

Reply:
Yeah, but that is for a separate refactoring PR that is not to be applied to older branches.

Reply:
`failed` provides a text describing for which type of credential there is a potential error.
Starting from line 1011, `failed` is non-NULL if and only if actually an error has been detected.